src/SFA.DAS.EAS.Web/Orchestrators/EmployerCommitmentsOrchestrator.cs

```diff
@@ -1102,8 +1102,8 @@ private Task<GetAccountTransferConnectionsResponse> GetTransferConnectionsNoAuth
             {
                 var legalEntity = await GetLegalEntityByCode(hashedAccountId, userId, legalEntityCode);
 
-                var hasSigned = legalEntity.AgreementStatus == EmployerAgreementStatus.Signed;
-
+                var hasSigned = HasSignedAgreement(legalEntity, !string.IsNullOrWhiteSpace(transferConnectionCode));
```

Comment:
As discussed, tranferConnectionCode comes from a hidden field on the webpage, so a malicious user could remove the contents to get past this check if they had only signed v1 of the agreement, but they have the option of creating a cohort anyway without having signed v2, and there is a hard check at the end of the journey.